### DISCUSSION TITLE
rtu_framer:247 Frame check failed, ignoring!!

### DISCUSSION THREAD

**Discussion by python926** [pymodbus-dev/pymodbus#1149] [Help] [answered]
running sync rtu modbus the straight code from the example and getting the error  
rtu_framer:247 Frame check failed, ignoring!!
how do I solve

**Comment by janiversen** [comment]
Without more information it is impossible to help you.
- which version are you using ?
- which example. ?

With 3.0.0 the directory examples (but not sub directories) are guaranteed to work because they are automatically tested.

**Reply by python926** [reply]
using version 3.0.0

**Reply by python926** [reply]
import asyncio
import logging

import pymodbus.diag_message as req_diag
import pymodbus.mei_message as req_mei
import pymodbus.other_message as req_other
from sync_client import run_sync_client, setup_rtu_sync_client
from pymodbus.exceptions import ModbusException
from pymodbus.pdu import ExceptionResponse
_logger = logging.getLogger()
SLAVE = 0x01
# -------------------------------------------------
# Generic error handling, to avoid duplicating code
# -------------------------------------------------
def _check_call(rr):
    """Check modbus call worked generically."""
    assert not rr.isError()  # test that call was OK
    assert not isinstance(rr, ExceptionResponse)  # Device rejected request
    return rr


def _handle_holding_registers(client):
    """Read/write holding registers."""
    _logger.info("### write holding register and read holding registers")
    #_check_call(await client.write_register(1, 10, slave=SLAVE))
    rr = _check_call(client.read_holding_registers(7000, 1, slave=SLAVE))
    #assert rr.registers[0] == 10



def template_call(client):
    """Show complete modbus call, sync version."""
    try:
        rr = client.read_holding_registers(7000, 1, slave=SLAVE)
    except ModbusException as exc:
        txt = f"ERROR: exception in pymodbus {exc}"
        _logger.error(txt)
        raise exc
    if rr.isError():
        txt = "ERROR: pymodbus returned an error!"
        _logger.error(txt)
        raise ModbusException(txt)
    if isinstance(rr, ExceptionResponse):
        txt = "ERROR: received exception from device {rr}!"
        _logger.error(txt)
        # THIS IS NOT A PYTHON EXCEPTION, but a valid modbus message
        raise ModbusException(txt)

    # Validate data
    txt = f"### Template coils response: {str(rr.bits)}"
    _logger.debug(txt)

def run_sync_calls(client):
    """Demonstrate basic read/write calls."""
    template_call(client)


if __name__ == "__main__":

    testclient = setup_rtu_sync_client(baud=19200, timeout=10,retries=3,port1='/dev/ttyUSB0')
    run_sync_client(testclient, modbus_calls=run_sync_calls)

**Comment by python926** [comment]
connecting with a usb to RS485 converter

**Comment by python926** [comment]
<bound method ModbusSerialClient.connect of <ModbusSerialClient at 0x7f90957f40 socket=None, framer=<pymodbus.framer.rtu_framer.ModbusRtuFramer object at 0x7f90203040>, timeout=10>>
12:52:23 DEBUG transaction:130 Current transaction state - IDLE
12:52:23 DEBUG transaction:134 Running transaction 1
12:52:23 DEBUG transaction:291 SEND: 0x1 0x3 0x1b 0x58 0x0 0x1 0x3 0x3d
12:52:23 DEBUG base:199 New Transaction state "SENDING"
12:52:23 DEBUG transaction:312 Changing transaction state from "SENDING" to "WAITING FOR REPLY"
12:52:24 DEBUG transaction:428 Changing transaction state from "WAITING FOR REPLY" to "PROCESSING REPLY"
12:52:24 DEBUG transaction:327 RECV: 0xe0 0x30 0xf8 0xaa 0xff 0xfa 0xee
12:52:24 DEBUG rtu_framer:247 Frame check failed, ignoring!!
12:52:24 DEBUG rtu_framer:132 Resetting frame - Current Frame in buffer - 0xe0 0x30 0xf8 0xaa 0xff 0xfa 0xee
12:52:24 DEBUG transaction:528 Getting transaction 1
12:52:24 DEBUG transaction:234 Changing transaction state from "PROCESSING REPLY" to "TRANSACTION_COMPLETE"
12:52:24 ERROR sync_call:62 ERROR: pymodbus returned an error!
Traceback (most recent call last):
  File "/home/pi/Scripts/sync_call.py", line 82, in <module>
    run_sync_client(testclient, modbus_calls=run_sync_calls)
  File "/home/pi/Scripts/sync_client.py", line 57, in run_sync_client
    modbus_calls(client)
  File "/home/pi/Scripts/sync_call.py", line 76, in run_sync_calls
    template_call(client)
  File "/home/pi/Scripts/sync_call.py", line 63, in template_call
    raise ModbusException(txt)
pymodbus.exceptions.ModbusException: Modbus Error: ERROR: pymodbus returned an error!

**Comment by janiversen** [comment, marked as answer]
I think we solved this problem, and it is available on dev.
